frontend/src/App.js

```diff
@@ -152,12 +170,20 @@ class App extends Component {
         let ingredients = [];
         // Find impact of each ingredient and rank them
         for (let item of json["items"]) {
+            let impact;
+            if (item["match_conf"] >= 0.5) {
+                impact = item["impact"] / 1000 * 2.2; // Convert from grams to pounds
+            } else {
+                impact = 0;
+            }
             ingredients.push({
-                name: item["product"]["alias"],
-                impact: item["impact"] / 1000 * 2.2, // Convert from grams to pounds
+                name: item["names"][0],
+                product: item["product"]["alias"],
+                impact: impact,
                 grams: item["g"]});
```

Comment:
You're assuming that GHGI gives you an item (even if it's a guess) when pushing the item, leading to errors. What if GHGI has no idea what item you're searching, returns null, and you try reading "alias" property of a non-existent item? 
And what "product" do you get? I see you already check and set "impact" according to "match_conf", but you need to do similar for product.

(Case 1) Possibility low match confidence, and incorrect product -> incorrect product recorded. Examples: "wasabi" 
[below: printed ingredients]
![image](https://user-images.githubusercontent.com/31525446/130293400-a0a2b948-1d48-4fe1-a648-4aefc7cadb5b.png)
I suppose your check for `Math.abs(item.impact) < 1e-5` in getting Land and Water use makes incorrectly labeled products not have any ill effect for now (since you set impact to 0), but might cause errors later on if we forget that and decide to change the code. Would suggest you set product name to something more accurate (like null), so if we ever try reading product impact, it raises an error instead of silently giving inaccurate data.

(Case 2) Possibility null match confidence, and null product -> leads to error. Examples: "asdfasf"
![image](https://user-images.githubusercontent.com/31525446/130293434-0eb00c58-f1c7-47cc-a8a9-dffbcd530770.png)